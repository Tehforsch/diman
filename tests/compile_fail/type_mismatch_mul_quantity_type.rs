#![allow(incomplete_features)]
#![feature(adt_const_params)]
#![feature(generic_const_exprs)]

use diman::{dimension, unit_system};

#[dimension]
pub struct Dimension {
    time: i32,
    length: i32,
}

#[rustfmt::skip]
unit_system!(
    Quantity,
    Dimension,
    [
        def Dimensionless = {},
        unit dimensionless = Dimensionless,
    ]
);

use crate::f64::*;

fn main() {
    let x: () = 1.0 * Dimensionless::dimensionless(1.0);
}
