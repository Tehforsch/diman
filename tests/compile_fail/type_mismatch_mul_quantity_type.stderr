error[E0308]: mismatched types
  --> tests/compile_fail/type_mismatch_mul_quantity_type.rs:26:17
   |
26 |     let x: () = 1.0 * Dimensionless::dimensionless(1.0);
   |            --   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `()`, found `Quantity<f64, Dimension>`
   |            |
   |            expected due to this
   |
   = note: expected unit type `()`
                 found struct `Quantity<f64, Dimension>`
